element_calcium_imaging/plotting/widget.py

```diff
@@ -0,0 +1,205 @@
+        ],
+        layout=fig2_layout,
+    )
+
+    fig1_widget = go.FigureWidget(fig1)
+    fig2_widget = go.FigureWidget(fig2)
+
+    def tooltip_click(trace, points, selector):
+        mask_id = trace.customdata[points.ys[0]][points.xs[0]]
+
+        if mask_id > -1:
+            activity_trace_figobj = from_json(
+                (
+                    ActivityReport
+                    & motioncorrection_dropdown.value
+                    & f"mask='{mask_id}'"
+                ).fetch1("activity_trace")
+            )
+
+            with fig2_widget.batch_update():
+                fig2_widget.data[0].x = activity_trace_figobj.data[0].x
+                fig2_widget.data[0].y = activity_trace_figobj.data[0].y
+                fig2_widget.data[0].name = activity_trace_figobj.data[0].name
+                fig2_widget.data[1].x = activity_trace_figobj.data[1].x
+                fig2_widget.data[1].y = activity_trace_figobj.data[1].y
+                fig2_widget.data[1].name = activity_trace_figobj.data[1].name
+                fig2_widget.layout["title"] = {
+                    "text": f"Trace for Cell {mask_id}",
+                    "xanchor": "center",
+                    "yanchor": "top",
+                    "y": 0.97,
+                    "x": 0.5,
+                }
+
+    def response(change, usedb=False):
+        if usedb:
+            composite_average_image = from_json(
+                (ScanLevelReport & motioncorrection_dropdown.value).fetch1(
+                    "average_image"
+                )
+            )
+
+            with fig1_widget.batch_update():
+                fig1_widget.data[0].z = composite_average_image.data[0].z
+                fig1_widget.data[0].customdata = composite_average_image.data[
+                    0
+                ].customdata
+
+                fig2_widget.data[0].x = None
+                fig2_widget.data[0].y = None
+                fig2_widget.data[1].x = None
+                fig2_widget.data[1].y = None
+        else:
+            image = (
+                imaging.MotionCorrection.Summary & motioncorrection_dropdown.value
+            ).fetch1("average_image")
+
+            cell_mask_ids, mask_xpix, mask_ypix = (
+                imaging.Segmentation.Mask * imaging.MaskClassification.MaskType
+                & motioncorrection_dropdown.value
+            ).fetch("mask", "mask_xpix", "mask_ypix")
+
+            background_image = cell_plot.single_to_3channel_image(image, low_q=0, high_q=99.9)
+
+
+            background_image_with_cells_painted = cell_plot.paint_rois(
+
+                background_image, mask_xpix, mask_ypix
+            )
+            cells_maskid_image = cell_plot.make_maskid_image(
+
+                background_image[:, :, 0], cell_mask_ids, mask_xpix, mask_ypix
+            )
+            background_with_cells = cell_plot.alpha_combine_2images(
+
+                background_image, background_image_with_cells_painted
+            )

```

Comment:
Can you refactor `plot_cell_overlayed_image` to also return these variables and thereby reduce the redundant code here?  Or split `plot_cell_overlayed_image` into 2 functions?

Reply:
I split the function into two.

Reply:
Thank you